sql/core/src/test/scala/org/apache/spark/sql/streaming/TransformWithStateSuite.scala

```diff
@@ -983,6 +983,40 @@ class TransformWithStateSuite extends StateStoreMetricsTest
       }
     }
   }
+
+  test("test that valid schema evolution changes column family id") {
```

Comment:
I am bit unsure about how we are going to deal with schema evolution for col family id here.
For this suite, let's say if we assign col family id `1` for `countState` in the first run. IIUC, after restart, in `StateCompatibilityChecker`, we will detect there is schema evolution and we will return a map of col family name to id as `countState` mapping to col family id `2`. Then if we call `get` on the new RocksDB statestore, we will encoding col family id `2` and try to read the old state - there will be none because the previous state rows are encoded as col family id `1`.

I feel like we should probably leave out the schema evolution portion for later; this seems outside the scope of this PR. Feel free to correct me if my above understanding is wrong.